sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -282,12 +292,23 @@ def _initialize(self, y, layer_units):
         self.coefs_ = []
         self.intercepts_ = []
 
-        for i in range(self.n_layers_ - 1):
+        for i in range(self.n_layers_ - 2):
             coef_init, intercept_init = self._init_coef(layer_units[i],
-                                                        layer_units[i + 1])
+                                                        layer_units[i + 1],
+                                                        self.activation_fncs[i]
+                                                        )
             self.coefs_.append(coef_init)
             self.intercepts_.append(intercept_init)
 
+        # for output layer, use the rule according to the activation function
+        # in the previous layer.
+        coef_init, intercept_init = self._init_coef(
```

Comment:
I think that you can also use `zip` and list concatenation to avoid temporary variable and append

Reply:
this one is tricky because it is only for the output layer.